crates/noirc_frontend/src/node_interner.rs

```diff
@@ -576,4 +582,23 @@ impl NodeInterner {
             is_test.then_some(*id)
         })
     }
+
+    pub fn add_method(
+        &mut self,
+        self_type: &Type,
+        method_name: String,
+        method_id: FuncId,
+    ) -> Option<FuncId> {
+        match self_type {
+            Type::Struct(struct_type, generics) => {
+                let key = (struct_type.borrow().id, method_name);
+                self.struct_methods.insert(key, (generics.clone(), method_id)).map(|(_, id)| id)
+            }
+            other => unreachable!("Tried adding method to non-struct type '{}'", other),
+        }
+    }
+
+    pub fn lookup_method(&self, id: StructId, method_name: &str) -> Option<FuncId> {
```

Comment:
This method also re doc comments

Reply:
I added a doc comment but I do not think it is useful here